deepmd/pt_expt/train/training.py

```
@@ -417,35 +386,69 @@ def fn(
-        nlist = _expand(nlist)
-        mapping = _expand(mapping)
-        fparam = _expand(fparam)
-        aparam = _expand(aparam)
-        charge_spin = _expand(charge_spin)
+    # Pad nf to a safe prime; keep real nloc and nall from the data.
+    #
+    # make_fx (tracing_mode="symbolic") unifies dimension symbols that share
+    # the same concrete value at trace time (duck-shape merging).  We take
+    # one frame ([:1]) to normalise nf, then pad it to a prime so PyTorch
+    # does not specialise it as the constant 1.  nloc and nall come from
+    # real data (typically 50–500 and 200–5000+), so they are already too
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)